backends/opentsdb/opentsdb.c

```diff
@@ -87,4 +87,118 @@ int process_opentsdb_response(BUFFER *b) {
     return discard_response(b, "opentsdb");
 }
 
+static inline void opentsdb_build_message(BUFFER *b, char *message, const char *hostname, int length) {
+    buffer_sprintf(
+            b
+            , "POST /api/put HTTP/1.1\r\n"
+              "Host: %s\r\n"
+              "Content-Type: application/json\r\n"
+              "Content-Length: %d\r\n"
+              "\r\n"
+              "%s"
+            , hostname
+            , length
+            , message
+    );
+}
 
+int format_dimension_collected_opentsdb_http(
+        BUFFER *b                 // the buffer to write data to
+        , const char *prefix        // the prefix to use
+        , RRDHOST *host             // the host this chart comes from
+        , const char *hostname      // the hostname (to override host->hostname)
+        , RRDSET *st                // the chart
+        , RRDDIM *rd                // the dimension
+        , time_t after              // the start timestamp
+        , time_t before             // the end timestamp
+        , BACKEND_OPTIONS backend_options // BACKEND_SOURCE_* bitmap
+) {
+    (void)host;
+    (void)after;
+    (void)before;
+
+    char message[1024];
+    char chart_name[RRD_ID_LENGTH_MAX + 1];
+    char dimension_name[RRD_ID_LENGTH_MAX + 1];
+    backend_name_copy(chart_name, (backend_options & BACKEND_OPTION_SEND_NAMES && st->name)?st->name:st->id, RRD_ID_LENGTH_MAX);
+    backend_name_copy(dimension_name, (backend_options & BACKEND_OPTION_SEND_NAMES && rd->name)?rd->name:rd->id, RRD_ID_LENGTH_MAX);
+
+    int length =  snprintfz(message
+                           , sizeof(message)
+                           ,"{"
+                            "\"metric\": \"%s.%s.%s\","
+                            "\"timestamp\": %llu,"
+                            "\"value\": "COLLECTED_NUMBER_FORMAT ","
+                            "\"tags\": {"
+                            "    \"host\": \"%s%s%s\""
+                            "  }"
+                            "}"
```

Comment:
Inconsistent indentation. Missing leading spaces for the third argument.

Reply:
For we send the JSON compact and readable, I began to use two tabs indentation, as you will see in the in the next commit.

Reply:
Cool! JSON is more readable now, indeed 👍. But what I meant was the parameter alignment:
```
                           , sizeof(message)
                           , "{"
                             "  \"metric\": \"%s.%s.%s\","
                             "  \"timestamp\": %llu,"
                             "  \"value\": "COLLECTED_NUMBER_FORMAT ","
                             "  \"tags\": {"
                             "    \"host\": \"%s%s%s\""
                             "  }"
                             "}"
                           , prefix
```
Look at the parameters `"..."`. I shifted them to the right by one space.

Reply:
All right, now I understood completely what you were saying! I copied you adjusts for the next pull!